### PR TITLE
Support for Distribution path mapping with AQL

### PR DESCRIPTION
**Pull request by YardenEdery**
- [x] All [tests](https://github.com/jfrog/jfrog-cli/CONTRIBUTING.md#tests) have passed. If this feature is not already covered by the tests, new tests have been added.
- [x] The pull request is targeting the `dev` branch.
- [x] The code has been validated to compile successfully by running `go vet ./...`.
- [x] The code has been formatted properly using `go fmt ./...`.

---
Added support of "pathMapping" property to file spec in order to have the ability to control where to distribute the artifacts that are being added dynamically when using AQL in the release bundle that is being created.

When looking for docker artifacts using AQL, the release bundle also includes, in the distribution server backend, all the layers for a manifest.json and all of the manifest.json files and their layers if its a list.manifest.json

From now on the "pathMapping" will take affect on those none specified added manifests and layers and will allow the users a native way of creating the file spec for distribution commands.

FYI This couldn't be achieved using "target" for the list.manifest.json example.

### CODE REVIEW

File: testdata/go/projectmissingdependency/projectmissingdependecy.go [hunk 1]
```diff
@@ -12,4 +12,4 @@ func execDependency() {
 func Exec() {
 	fmt.Println("Executing ")
 	execDependency()
-}
\ No newline at end of file
+}
```
Comment (eyalbe4) [inline testdata/go/projectmissingdependency/projectmissingdependecy.go:15]: Unnecessary change which adds a redundant new-line.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
* Let's add a complete description to this PR, explaiining the new functionality,
* Let's create a documentation PR to https://github.com/jfrog/documentation. This documentation repository include the source content for JFrog CLI's online documentation available at https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli.
I suggest adding a few examples to https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-distribution and also updating https://docs.jfrog-applications.jfrog.io/jfrog-applications/jfrog-cli/cli-for-jfrog-artifactory#using-file-specs

File: testdata/go/projectmissingdependency/projectmissingdependecy.go (hunk 1 above)
Comment (YardenEdery) [reply testdata/go/projectmissingdependency/projectmissingdependecy.go:15]: But gofmt makes the go file end with a newline, are you sure not to align this on current file?

**Comment by YardenEdery** [comment]
Should we also create documentation in:
https://jfrog.com/help/r/jfrog-cli/cli-for-jfrog-artifactory

File: schema/filespec-schema.json [hunk 2]
```diff
@@ -42,6 +42,19 @@
           }
         }
       },
+      "pathMapping": {
+        "description": "If specified, determines release-bundle source and target paths for artifacts that had been fetched using aql",
```
Comment (yahavi) [inline schema/filespec-schema.json:46]:
```suggestion
        "description": "If specified, determines release-bundle source and target paths for artifacts that had been fetched using aql.",
```

File: schema/filespec-schema.json [hunk 3]
```diff
@@ -42,6 +42,19 @@
...
+        "properties": {
+          "input": {
+            "type": "string",
+            "description": "The input path for mapping."
```
Comment (yahavi) [inline schema/filespec-schema.json:50]: Configuring `pathMapping` is challenging and the documentation does not explain how to do it. Let's add an example:
```suggestion
            "description": "The input path for mapping.",
            "examples": ["(.*)/old_folder/(.*)"]
```

File: schema/filespec-schema.json [hunk 4]
```diff
@@ -42,6 +42,19 @@
...
+          },
+          "output": {
+            "type": "string",
+            "description": "The output path for mapping."
```
Comment (yahavi) [inline schema/filespec-schema.json:54]:
```suggestion
            "description": "The output path for mapping.",
            "examples": ["$1/new_folder/$2"]
```

**Review by yahavi — CHANGES_REQUESTED** [review]

**Comment by yahavi** [comment]
Please also update `go.mod` and `go.sum` with the tip of dev branch to reflect the changes in https://github.com/jfrog/jfrog-cli-core/pull/1031
